distributed/tests/test_s3.py

```diff
@@ -27,60 +27,171 @@
+    assert fn not in s3.ls('distributed-test/')
+    assert fn in s3.ls('distributed-test/nested/')
+    assert fn in s3.ls('distributed-test/nested')
+    assert fn in s3.ls('s3://distributed-test/nested/')
 
-def test_read_keys_from_bucket():
-    for k, data in files.items():
-        file_contents = read_content_from_keys('distributed-test', k, anon=True)
 
-        assert file_contents == data
+def test_s3_glob(s3):
+    fn = 'distributed-test/nested/file1'
+    assert fn not in s3.glob('distributed-test/')
+    assert fn not in s3.glob('distributed-test/*')
+    assert fn in s3.glob('distributed-test/nested')
+    assert fn in s3.glob('distributed-test/nested/*')
+    assert fn in s3.glob('distributed-test/nested/file*')
+    assert fn in s3.glob('distributed-test/*/*')
 
-    assert (read_content_from_keys('s3://distributed-test', k, anon=True) ==
-            read_content_from_keys('distributed-test', k, anon=True))
 
+def test_get_list_of_summary_objects(s3):
+    L = s3.ls(test_bucket_name + '/test')
 
-def test_list_summary_object_with_prefix_and_delimiter():
-    keys = get_list_of_summary_objects(test_bucket_name, 'nested/nested2/',
-                                       delimiter='/', anon=True)
+    assert len(L) == 2
+    assert [l.lstrip(test_bucket_name).lstrip('/') for l in sorted(L)] == sorted(list(files))
 
-    assert len(keys) == 2
-    assert [k.key for k in keys] == [u'nested/nested2/file1',
-                                     u'nested/nested2/file2']
+    L2 = s3.ls('s3://' + test_bucket_name + '/test')
 
-    keys = get_list_of_summary_objects(test_bucket_name, 'nested/', anon=True)
+    assert L == L2
 
-    assert len(keys) == 4
-    assert [k.key for k in keys] == [u'nested/file1',
-                                     u'nested/file2',
-                                     u'nested/nested2/file1',
-                                     u'nested/nested2/file2']
 
+def test_read_keys_from_bucket(s3):
+    for k, data in files.items():
+        file_contents = s3.cat('/'.join([test_bucket_name, k])) 
+        assert file_contents == data
 
-@gen_cluster(timeout=60, executor=True)
-def test_read_bytes(e, s, a, b):
-    futures = read_bytes(test_bucket_name, prefix='test/', anon=True,
-                         lazy=False)
+    assert (s3.cat('/'.join([test_bucket_name, k])) ==
+            s3.cat('s3://' + '/'.join([test_bucket_name, k])))
+
+
+def test_seek_delimiter(s3):
+    fn = 'test/accounts.1.json'
+    data = files[fn]
+    with s3.open('/'.join([test_bucket_name, fn])) as f:
+        seek_delimiter(f, b'}', 0)
+        assert f.tell() == 0
+        f.seek(1)
+        seek_delimiter(f, b'}', 5)
+        assert f.tell() == data.index(b'}') + 1
+        seek_delimiter(f, b'\n', 5)
+        assert f.tell() == data.index(b'\n') + 1
+        f.seek(1, 1)
+        ind = data.index(b'\n') + data[data.index(b'\n')+1:].index(b'\n') + 1
+        seek_delimiter(f, b'\n', 5)
+        assert f.tell() == ind + 1
+
+
```

Comment:
This test is slow.  We should decorate it with the `slow` decorator in `utils_test.py`.

Alternatively, it would be interesting to learn to use `moto`, which would really help with things like this.  If you want to dive into S3 more seriously it might be worth investing some time here.